deviceModeInit.js

```diff
@@ -149,6 +160,18 @@ var rudderTracking = (function () {
     return pageURL.indexOf(name) > -1 ? true : false;
   }
 
+  // fire a track call only when both /collections and /products are present
+  // i.e at URL /collections/{collectionName}/products
+
```

Comment:
This logic will fail for 
`/collections/all` and such kinds of URL signature

Reply:
Rather than relying on URL signature, lets use the response of
`url = ${referrer}products.json;` to actual backend to verify if we are on a single products page or a products list page and take action accordingly as discussed

Reply:
If the current implementation resolves the defect at hand, let's proceed. The tracking logic in this file needs to be revamped to resolve other defects as well.

Reply:
Actually, it is not failing since for /collections/... it is being handled in the new implementation.